src/java.base/share/classes/sun/nio/ch/FileChannelImpl.java

```diff
@@ -779,9 +798,12 @@ public long transferFrom(ReadableByteChannel src,
             throw new IllegalArgumentException();
         if (position > size())
             return 0;
-        if (src instanceof FileChannelImpl)
-           return transferFromFileChannel((FileChannelImpl)src,
-                                          position, count);
+
+        if (src instanceof FileChannelImpl) {
+            long n = transferFromFileChannel((FileChannelImpl)src, position, count);
```

Comment:
It might be a bit cleaner to use the pattern matching with the instanceof to avoid the cast to FileChannelImpl.

Reply:
Right, I should have caught that myself.